x-pack/plugins/infra/public/pages/logs/log_entry_rate/sections/anomalies/index.tsx

```diff
@@ -97,52 +77,44 @@ export const AnomaliesResults: React.FunctionComponent<{
         </EuiFlexItem>
       </EuiFlexGroup>
       <EuiSpacer size="m" />
-      {(!logEntryRateResults ||
-        (logEntryRateResults &&
-          logEntryRateResults.histogramBuckets &&
-          !logEntryRateResults.histogramBuckets.length)) &&
-      (!anomalies || anomalies.length === 0) ? (
-        <LoadingOverlayWrapper
-          isLoading={isLoadingLogRateResults || isLoadingAnomaliesResults}
-          loadingChildren={<LoadingOverlayContent />}
-        >
-          <EuiEmptyPrompt
-            title={
-              <h2>
-                {i18n.translate('xpack.infra.logs.analysis.anomalySectionNoDataTitle', {
-                  defaultMessage: 'There is no data to display.',
-                })}
-              </h2>
-            }
-            titleSize="m"
-            body={
-              <p>
-                {i18n.translate('xpack.infra.logs.analysis.anomalySectionNoDataBody', {
-                  defaultMessage: 'You may want to adjust your time range.',
-                })}
-              </p>
-            }
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <Visualisation
```

Comment:
As already discussed elsewhere, putting both the swimlane and the table, which each have their own pagination, into the same panel can be very confusing:

![image](https://user-images.githubusercontent.com/973741/106890216-4a022600-66e9-11eb-8b71-602f1729981f.png)


There is no way for the user to anticipate that the highlighted pagination won't influence the table that follows right after it.

Putting each into their own panel or otherwise separating them such that the association of the pagination controls to their respective part of the UI becomes clear. This could be done in a follow-up PR.

Reply:
Yep. Copying ML themselves will probably be best (two panels, like you say).